src/test/java/teammates/ui/webapi/DeadlineExtensionsWorkerActionTest.java

```diff
@@ -0,0 +1,181 @@
+        assertEquals(newEndTime2,
+                logic.getDeadlineExtension(courseId, feedbackSessionName, studentUpdate.getEmail(), false).getEndTime());
+        verifyAbsentInDatabase(studentExtensionToDelete);
+        assertEquals(newEndTime1,
+                logic.getDeadlineExtension(courseId, feedbackSessionName, instructorCreate.getEmail(), true).getEndTime());
+        assertEquals(newEndTime2,
+                logic.getDeadlineExtension(courseId, feedbackSessionName, instructorUpdate.getEmail(), true).getEndTime());
+        verifyAbsentInDatabase(instructorExtensionToDelete);
+
+        verifySpecifiedTasksAdded(Const.TaskQueue.SEND_EMAIL_QUEUE_NAME, 6);
+
+        List<TaskWrapper> tasksAdded = mockTaskQueuer.getTasksAdded();
+        String datetimeDisplayFormat = "EEE, dd MMM yyyy, hh:mm a z";
+        for (TaskWrapper task : tasksAdded) {
+            SendEmailRequest requestBody = (SendEmailRequest) task.getRequestBody();
+            EmailWrapper email = requestBody.getEmail();
+            String expectedSubject;
+
+            if (email.getRecipient().equals(studentCreate.getEmail())
+                    || email.getRecipient().equals(instructorCreate.getEmail())) {
+                expectedSubject = String.format(
+                        EmailType.DEADLINE_EXTENSION_GIVEN.getSubject(), courseName, feedbackSessionName);
+                assertEquals(expectedSubject, email.getSubject());
+                assertTrue(email.getContent().contains(
+                        TimeHelper.formatInstant(newEndTime1, "Africa/Johannesburg", datetimeDisplayFormat)));
+                continue;
+            }
+
+            if (email.getRecipient().equals(studentRevoke.getEmail())
+                    || email.getRecipient().equals(instructorRevoke.getEmail())) {
+                expectedSubject = String.format(
+                        EmailType.DEADLINE_EXTENSION_REVOKED.getSubject(), courseName, feedbackSessionName);
+                assertEquals(expectedSubject, email.getSubject());
+                assertTrue(email.getContent().contains(TimeHelper.formatInstant(originalEndTime,
+                        "Africa/Johannesburg", datetimeDisplayFormat)));
+                continue;
+            }
+
+            expectedSubject = String.format(
+                    EmailType.DEADLINE_EXTENSION_UPDATED.getSubject(), courseName, feedbackSessionName);
+            assertEquals(expectedSubject, email.getSubject());
+            assertTrue(email.getContent().contains(
+                        TimeHelper.formatInstant(newEndTime2, "Africa/Johannesburg", datetimeDisplayFormat)));
+        }
+
+        ______TS("notifyUsers set to false; no tasks added");
+
+        logic.deleteDeadlineExtension(courseId, feedbackSessionName, studentCreate.getEmail(), false);
+        logic.deleteDeadlineExtension(courseId, feedbackSessionName, instructorCreate.getEmail(), true);
+        removeAndRestoreTypicalDataBundle();
```

Comment:
Why delete extension before `removeAndRestore`?

Also, after these are done, deadlines for `studentCreate` and `instructorCreate` are both the `newEndTime`?

Reply:
>Why delete extension before removeAndRestore?

My bad, made some wrong assumptions here. 

> Also, after these are done, deadlines for studentCreate and instructorCreate are both the newEndTime?

`studentCreate` and `instructorCreate` weren't modified. I don't believe there will be any changes.